File: benchmarks/competitors/prime_count.c
#include <stdio.h>

static long long count_primes(long long limit) {
    long long count = 0;
    for (long long number = 2; number < limit; number++) {
        int prime = 1;
        for (long long divisor = 2; divisor * divisor <= number; divisor++) {
            if (number % divisor == 0) {
                prime = 0;
                break;
            }
        }
        if (prime) {
            count++;
        }
    }
    return count;
}

int main(void) {
    printf("primes below 400000: %lld\n", count_primes(400000));
    return 0;
}

File: benchmarks/competitors/prime_count.rs
fn count_primes(limit: i64) -> i64 {
    let mut count = 0i64;
    let mut number = 2i64;
    while number < limit {
        let mut prime = true;
        let mut divisor = 2i64;
        while divisor * divisor <= number {
            if number % divisor == 0 {
                prime = false;
                break;
            }
            divisor += 1;
        }
        if prime {
            count += 1;
        }
        number += 1;
    }
    count
}

fn main() {
    println!("primes below 400000: {}", count_primes(400_000));
}
